src/core/models.py

```diff
@@ -1030,30 +1031,29 @@ def formats(self):
     def chapters(self):
         return self.chapter_set.all()
 
-    def onetaskers(self):
-        copyedit_assignments = CopyeditAssignment.objects.filter(book=self)
-        index_assignments = IndexAssignment.objects.filter(book=self)
-        typeset_assignments = TypesetAssignment.objects.filter(book=self)
-        review_assignments = ReviewAssignment.objects.filter(book=self)
-        users = []
-
-        for assignment in copyedit_assignments:
-            user = assignment.copyeditor
-            if user not in users:
-                users.append(user)
-        for assignment in index_assignments:
-            user = assignment.indexer
-            if user not in users:
-                users.append(user)
-        for assignment in typeset_assignments:
-            user = assignment.typesetter
-            if user not in users:
-                users.append(user)
-        for assignment in review_assignments:
-            user = assignment.user
-            if user not in users:
-                users.append(user)
-        return users
+    def get_onetaskers(self):
+        return set(
+            [
+                assignment.copyeditor
+                for assignment in CopyeditAssignment.objects.filter(book=self)
+            ] + [
+                assignment.indexer for assignment in
+                IndexAssignment.objects.filter(book=self)
+            ] + [
+                assignment.typesetter for assignment in
+                TypesetAssignment.objects.filter(book=self)
+            ] + [
+                assignment.user for assignment in
+                ReviewAssignment.objects.filter(book=self)
+            ] + [
+                review.user for review in
+                er_models.EditorialReview.objects.filter(
+                    object_id=self.id,
+                    content_type__model='book',
+                )
+            ]
+        )
+
```

Comment:
set(item for sublist in [[1],[2]] for item in sublist) no need for that clumsy concatenation using the `+` operator

https://stackoverflow.com/questions/952914/making-a-flat-list-out-of-list-of-lists-in-python

Reply:
or maybe https://docs.python.org/2.7/library/itertools.html#itertools.chain.from_iterable (also mentioned a bit further down in that post.)

Reply:
Though I have a feeling this could also be done using the Django ORM as you did below (since you are just collecting a set of users?)

Reply:
I've opted for `itertools.chain` over the awkward nested list comprehensions that would be required for your suggestion. Thanks for pointing me in the right direction, though.

Reply:
> or maybe https://docs.python.org/2.7/library/itertools.html#itertools.chain.from_iterable (also mentioned a bit further down in that post.)

Ah, hadn't refreshed my page - yeah `chain` is a better way of doing it. And yes, I also rewrote most of it using the ORM. For the Django 2 version of rua, I'll use `Queryset.union` and avoid `itertools` alltogether.